argo_jupyter_scheduler/executor.py

```diff
@@ -573,3 +663,143 @@ def create_job_record(
 
         session.add(job)
         session.commit()
+
+
+@script()
+def rename_files(db_url, job_definition_id, input_path, log_path, start_time):
+    import os
+
+    from jupyter_scheduler.orm import Job, create_session
+
+    from argo_jupyter_scheduler.utils import (
+        add_file_logger,
+        gen_default_html_path,
+        gen_default_output_path,
+        gen_html_path,
+        gen_output_path,
+        setup_logger,
+    )
+
+    try:
+        # Sets up logging
+        logger = setup_logger("rename_files")
```

Comment:
This shadows name `logger` from [outer scope.](https://github.com/nebari-dev/argo-jupyter-scheduler/pull/3/files#diff-a94f5b21a4c546e0bc773f05eb530063baba8f9a5f869b157c9a1cf230bd92c3R32)

Reply:
Nope, that's on purpose. I've just re-tested to be sure, too. The global `logger` name is not accessible here. Things used in these `script`s need to be local to them because they'll be running as separate pods. That's why they have these local imports. And you also cannot pass arbitrary Python objects as parameters - only very basic serializable things, like strings and dicts.